### PR TITLE
Edit student rep ld

### PR DESCRIPTION
**Pull request by LoganDemaray316**
Student can now edit and delete their pieces in a repository.

Edited Files:
RepertoireComponent.vue
StudentHomeDashboard.vue
UserStore.js

Created Files:
RepertoireEdit.vue

### CODE REVIEW

File: src/stores/UserStore.js [hunk 1]
```diff
@@ -340,12 +339,55 @@ export const useUserStore = defineStore("user", {
...
           (repertoire) => repertoire.repertoireId === data.repertoireId
         )[0]
         .pieces.push(pieceInfo);
     },
+    async editPiece(data, id) {
+      let pieceInfo = data;
+      await PieceDataService.update(id, data).catch((e) => {
+        console.log(e);
+      });
+      let composerInfo = await this.getComposerInfo(data.composerId);
```
Comment (ethanimooney) [inline src/stores/UserStore.js:354]: You could add .Composers[0] to the end of this so you don't have to do it every time below

**Review by ethanimooney — APPROVED** [review]
Just a few little things, LGTM otherwise.

File: src/stores/UserStore.js (hunk 1 above)
Comment (LoganDemaray316) [reply src/stores/UserStore.js:354]: Done

File: src/components/Student/RepertoireEdit.vue [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            data = response.data.Instruments;
+          })
+          .catch((e) => {
+            console.log(e);
+          });
+        return data;
+      },
+      async getComposers() {
+        let data = [];
+        await ComposersDataService.getAll()
```
[the remark this answers is not in the document]
Comment (LoganDemaray316) [reply src/components/Student/RepertoireEdit.vue:187]: Ok

File: src/components/Student/RepertoireEdit.vue [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    <v-btn @click="createDialog = true" elevation="0" class="text-darkBlue">
+      Missing some information? Request it here.
+    </v-btn>
+    <v-dialog v-model="createDialog" persistent max-width="600px">
+      <MissingInformation
+        @closeCourseDialogEvent="closeCreateDialog"></MissingInformation>
+    </v-dialog>
+    <v-card-actions class="mx-auto font-weight-bold">
+      <v-btn @click="closeAndAddDialog()" color="darkBlue">Save</v-btn>
+      <v-btn @click="deletePiece()" color="red">Delete</v-btn>
```
[the remark this answers is not in the document]
Comment (LoganDemaray316) [reply src/components/Student/RepertoireEdit.vue:97]: Done

File: src/stores/UserStore.js [hunk 4]
```diff
@@ -340,12 +339,55 @@ export const useUserStore = defineStore("user", {
...
+        console.log(e);
+      });
+      this.userRoleInfo.repertoires
+        .filter((repertoire) => repertoire.repertoireId === repertoireId)[0]
+        .pieces.splice(
+          this.userRoleInfo.repertoires
+            .filter((repertoire) => repertoire.repertoireId === repertoireId)[0]
+            .pieces.findIndex((piece) => piece.pieceId === id),
+          1
+        );
```
[the remark this answers is not in the document]
Comment (LoganDemaray316) [reply src/stores/UserStore.js:389]: Done

**Pull request merged**